src/com/serotonin/mango/db/MySQLAccess.java

```diff
@@ -116,33 +116,55 @@ protected boolean newDatabaseCheck(ExtendedJdbcTemplate ejt) {
         	baseLineNotExist = true;
         	LOG.info("BaseLineNotExist:"+baseLineNotExist);
         }
-        
+
+        if (shemaExist && !baseLineNotExist) {
+            CorrectMigration2_2.correct();
+        }
+
+
         try {
-	        Flyway flyway = new Flyway();
-			flyway.setLocations("org.scada_lts.dao.migration.mysql");
-			flyway.setDataSource(getDataSource());
-			
-	        if (shemaExist) {
-	        	// old shema without flayway
-	        	if (baseLineNotExist) {
-	        		flyway.setBaselineOnMigrate(true);
-	    			flyway.baseline();
-	    			flyway.migrate();
-	    		}
-	        } else {
-	        	//shema not exist
-	        	if (baseLineNotExist) {
-	    			//flyway.baseline();
-	    			flyway.migrate();		
-	    		}
-	        }
-	        
-	        flyway.migrate();
-	        
-        } catch (FlywayException fe) {
-        	LOG.error(fe);
+            Flyway flyway = null;
```

Comment:
Here you can do refactoring:

```
flyway = null;

if(shemaExist) {

	if(baseLineNotExist) {
		flyway = A; //shemaExist == true && baseLineNotExist == true
		flyway.migrate();
	}

} else {

	if(baseLineNotExist) {
		flyway = B;  //shemaExist == false && baseLineNotExist == true
		flyway.migrate();
	}
}

if(flyway == null) {
	flyway = C;
}

flyway.migrate();

```
=

```
flyway = null;

if(shemaExist && baseLineNotExist) {
	flyway = A;
	flyway.migrate();
}

if(!shemaExist && baseLineNotExist) {
	flyway = B;
	flyway.migrate();
}

if(flyway == null) {
	flyway = C;
}

flyway.migrate();

```
=

```
flyway = null;

if(baseLineNotExist) {
	if(shemaExist) {
		flyway = A; //not null
		flyway.migrate();
	} else {
		flyway = B; //not null
		flyway.migrate();
	}
}

if(flyway == null) {
	flyway = C;
}

flyway.migrate();
```

Because flyway is null only if baseLineNotExist == false :

```
flyway = null;

if(baseLineNotExist) {
	if(shemaExist) {
		flyway = A;
		flyway.migrate();
	} else {
		flyway = B;
		flyway.migrate();
	}
} else {
	flyway = C;
}

flyway.migrate();
```

so:

```
Flyway flyway = null;

if(baseLineNotExist) {
	if(shemaExist) {
	
		flyway = Flyway.configure()
				.baselineOnMigrate(true)
				.dataSource(getDataSource())
				.locations("org.scada_lts.dao.migration.mysql")
				.table("schema_version")
				.load();

		flyway.baseline();
		flyway.migrate();
		
	} else {
	
		flyway = Flyway.configure()
				.baselineOnMigrate(true)
				.dataSource(getDataSource())
				.locations("org.scada_lts.dao.migration.mysql")
				.table("schema_version")
				.load();
		flyway.migrate();
	}
} else {

	flyway = Flyway.configure()
		.dataSource(getDataSource())
		.locations("org.scada_lts.dao.migration.mysql")
		.table("schema_version")
		.load();
	
}

flyway.migrate();
```

call:

`.baselineOnMigrate(true)`

is true or false, and if baseLineNotExist == true then true, so:

```
Flyway flyway = Flyway.configure()
				.baselineOnMigrate(baseLineNotExist)
				.dataSource(getDataSource())
				.locations("org.scada_lts.dao.migration.mysql")
				.table("schema_version")
				.load();

if(baseLineNotExist) {
        if(shemaExist) {
              flyway.baseline();
        }
	flyway.migrate();
}

flyway.migrate();
```

if baseLineNotExist == true then twice calls migrate() on the same Flyway object, is it good? If not:

```
Flyway flyway = Flyway.configure()
				.baselineOnMigrate(baseLineNotExist)
				.dataSource(getDataSource())
				.locations("org.scada_lts.dao.migration.mysql")
				.table("schema_version")
				.load();
				
if(baseLineNotExist && shemaExist) {
	flyway.baseline();
} 

flyway.migrate();
```